### PR TITLE
GPU: Revise swapchain acquisition

### PR DESCRIPTION
**Pull request by thatcosmonaut**
- Adds SDL_WaitForGPUSwapchain() which blocks until a swapchain texture is available
- Changes SDL_AcquireGPUSwapchainTexture() to immediately return and fill the texture pointer with null if too many frames are in flight regardless of present mode
- Adds SDL_WaitAndAcquireGPUSwapchainTexture() which blocks until a swapchain texture is available and then acquires it

This should allow clients to effectively choose between throughput-optimized and latency-optimized workflows.

Supersedes #11624

### CODE REVIEW

**Comment by thatcosmonaut** [comment]
Looks like the test failures are coming from main.

File: src/gpu/d3d12/SDL_gpu_d3d12.c [hunk 1]
```diff
@@ -7124,7 +7124,32 @@ static SDL_GPUCommandBuffer *D3D12_AcquireCommandBuffer(
     return (SDL_GPUCommandBuffer *)commandBuffer;
 }
 
-static bool D3D12_AcquireSwapchainTexture(
+static bool D3D12_WaitForSwapchain(
+    SDL_GPURenderer *driverData,
+    SDL_Window *window)
+{
+    D3D12Renderer *renderer = (D3D12Renderer *)driverData;
```
Comment (kg) [inline src/gpu/d3d12/SDL_gpu_d3d12.c:7131]: unused? and not null checked

File: src/gpu/SDL_gpu.c [hunk 2]
```diff
@@ -2725,6 +2725,63 @@ bool SDL_AcquireGPUSwapchainTexture(
...
 }
 
+bool SDL_WaitForGPUSwapchain(
+    SDL_GPUDevice *device,
+    SDL_Window *window)
+{
+    CHECK_DEVICE_MAGIC(device, false);
+
+    if (window == NULL) {
+        SDL_InvalidParamError("window");
```
Comment (slouken) [inline src/gpu/SDL_gpu.c:2735]:
```suggestion
        return SDL_InvalidParamError("window");
```

File: src/gpu/SDL_gpu.c [hunk 3]
```diff
@@ -2725,6 +2725,63 @@ bool SDL_AcquireGPUSwapchainTexture(
...
+    SDL_GPUCommandBuffer *command_buffer,
+    SDL_Window *window,
+    SDL_GPUTexture **swapchain_texture,
+    Uint32 *swapchain_texture_width,
+    Uint32 *swapchain_texture_height)
+{
+    CommandBufferCommonHeader *commandBufferHeader = (CommandBufferCommonHeader *)command_buffer;
+
+    if (command_buffer == NULL) {
+        SDL_InvalidParamError("command_buffer");
```
Comment (slouken) [inline src/gpu/SDL_gpu.c:2754]:
```suggestion
        return SDL_InvalidParamError("command_buffer");
```
... etc.

**Comment by slouken** [comment]
In general this looks good. Poking @icculus about the build failures...

**Comment by icculus** [comment]
Build failure is unrelated; I just pushed a fix for it. This PR can merge once everyone is happy with it.

File: src/gpu/d3d12/SDL_gpu_d3d12.c (hunk 1 above)
Comment (thatcosmonaut) [reply src/gpu/d3d12/SDL_gpu_d3d12.c:7131]: It's actually used in the macro

**Pull request merged**